Decouple Tempo frontend

fabrizio-grafana opened the pull request:
_Ready for review, although we miss the `@grafana/traces` package (name not yet definitive) to have this work properly._

Decoupling of Tempo frontend (TypeScript code) from Grafana. 

When possible and convenient, imports of Grafana internally modules (Grafana core, utils, etc.) have been replaced with Grafana packages or third-part packages. Instead, Grafana modules such as those related to Loki and Prometheus that we didn't replace with external modules have been duplicated from Grafana into the `_importedDependencies` folder. This should allow us to easily keep track of what are the modules that in the long-term future should be removed from the Tempo plugin, for instance once the Loki and Prometheus plugins are available.

Some files and changes for the build and release process are also added, although the setup is not complete, since we also need the migration of the backend. For this reason, please note that this PR is currently not targeting `main` but a feature branch that we can use to bundle together the migration of the backend and frontend, avoiding to merge them separately in `main`.

fabrizio-grafana committed:
Decouple Tempo frontend

fabrizio-grafana committed:
Merge branch 'main' into otp/decouple-tempo-frontend

fabrizio-grafana committed:
Revert rename of plugin

fabrizio-grafana committed:
Merge branch 'main' into otp/decouple-tempo-frontend

ashharrison90 committed:
Chore: Improve typings (#78940)

* improve typings

* transformations typings improvements

* add bootData to window

* more grafana-data fixes

* undo window.grafanaBootData typings

joey-grafana committed:
Pyroscope: Decouple config editor (#79027)

Decouple config editor

mgyongyosi committed:
Auth: Fix for the github_oauth parse config error (#79063)

* Fix for reverting the github_oauth parse config behaviour

* minimal changes

alexanderzobnin committed:
Team LBAC: Limit access to data source (#78376)

* Team LBAC: Limit access to data source

* Fix tests

dprokop committed:
QueryVariable: Always run queries with dashboard time range (#79056)

toddtreece committed:
Revert "K8s: Refactor metrics to share k8s registry (#77957)" (#79079)

This reverts commit 38bc41651a416bd547adbb284bc1559a5aad1006.

DanCech committed:
Server: Automatically generate a self-signed TLS cert if needed (#78243)

generate a self-signed cert when running in https or http2 mode without a provided cert

alexanderzobnin committed:
Folders: Show dashboards and folders with directly assigned permissions in "Shared" folder (#78465)

* Folders: Show folders user has access to at the root level

* Refactor

* Refactor

* Hide parent folders user has no access to

* Skip expensive computation if possible

* Fix tests

* Fix potential nil access

* Fix duplicated folders

* Fix linter error

* Fix querying folders if no managed permissions set

* Update benchmark

* Add special shared with me folder and fetch available non-root folders on demand

* Fix parents query

* Improve db query for folders

* Reset benchmark changes

* Fix permissions for shared with me folder

* Simplify dedup

* Add option to include shared folder permission to user's permissions

* Fix nil UID

* Remove duplicated folders from shared list

* Folders: Fix fetching empty folder

* Nested folders: Show dashboards with directly assigned permissions

* Fix slow dashboards fetch

* Refactor

* Fix cycle dependencies

* Move shared folder to models

* Fix shared folder links

* Refactor

* Use feature flag for permissions

* Use feature flag

* Review comments

* Expose shared folder UID through frontend settings

* Add frontend type for sharedWithMeFolderUID option

* Refactor: apply review suggestions

* Fix parent uid for shared folder

* Fix listing shared dashboards for users with access to all folders

* Prevent creating folder with "shared" UID

* Add tests for shared folders

* Add test for shared dashboards

* Fix linter

* Add metrics for shared with me folder

* Add metrics for shared with me dashboards

* Fix tests

* Tests: add metrics as a dependency

* Fix access control metadata for shared with me folder

* Use constant for shared with me

* Optimize parent folders access check, fetch all folders in one query.

* Use labels for metrics

Jguer committed:
Plugins: Use org role to separate unauthed from anon (#79082)

use org role to separate unauthed from anon

ashharrison90 committed:
PluginList: Add plugin list loading skeleton (#79012)

* add plugin skeleton

* lineHeight: 1 instead of 0

gotjosh committed:
Unified Alerting: Set `max_attempts` to 1 by default (#79095)

* Unified Alerting: Set `max_attempts` to 1 by default

The retry logic for unified alerting has been broken as far as v9.4.x, rather than fixing it in one go and causing a headache to our users with rules putting extra load on their datasources - I think a better approach is to simply set 1 as a default and then let our users change it.

I see two cons with this approach:

- Configuration for legacy to unified alerting cannot be ported over automatically, users will have to manually set `max_attempts` to 3 when migrating.
- Users expecting to get any sort of retrying (as with legacy alerting) will not have it out of the box and will have to manually edit the configuration.

Signed-off-by: gotjosh <josue.abreu@gmail.com>
---------

Signed-off-by: gotjosh <josue.abreu@gmail.com>

evictorero committed:
Chore: Use dashboard service to retrieve dashboards (#79020)

ryantxu committed:
K8s: fix standalone command and add hack scripts (#79052)

Co-authored-by: Charandas Batra <charandas.batra@grafana.com>

zoltanbedi committed:
Chore: Add comments to RAQB config (#79075)

gabor committed:
sql: remove `xorm` dependency from postgres/mysql/mssql (#77870)

* sql: remove xorm

* sql: remove pkg/util dependency (#78821)

sql: removed md5-util dependency

ryantxu committed:
Chore: remove unused math function (#79120)

axelavargas committed:
Dashboard: Scene Migration: Add variable `definition` prop to serialization (#79022)

Co-authored-by: Ivan Ortega Alba <ivanortegaalba@gmail.com>

itsmylife committed:
InfluxDB: Implement InfluxQL json streaming parser (#76934)

* Have the first iteration

* Prepare bench testing

* rename the test files

* Remove unnecessary test file

* Introduce influxqlStreamingParser feature flag

* Apply streaming parser feature flag

* Add new tests

* More tests

* return executedQueryString only in first frame

* add frame meta and config

* Update golden json files

* Support tags/labels

* more tests

* more tests

* Don't change original response_parser.go

* provide context

* create util package

* don't pass the row

* update converter with formatted frameName

* add executedQueryString info only to first frame

* update golden files

* rename

* update test file

* use pointer values

* update testdata

* update parsing

* update converter for null values

* prepare converter for table response

* clean up

* return timeField in fields

* handle no time column responses

* better nil field handling

* refactor the code

* add table tests

* fix config for table

* table response format

* fix value

* if there is no time column set name

* linting

* refactoring

* handle the status code

* add tracing

* Update pkg/tsdb/influxdb/influxql/converter/converter_test.go

Co-authored-by: İnanç Gümüş <m@inanc.io>

* fix import

* update test data

* sanity

* sanity

* linting

* simplicity

* return empty rsp

* rename to prevent confusion

* nullableJson field type for null values

* better handling null values

* remove duplicate test file

* fix healthcheck

* use util for pointer

* move bench test to root

* provide fake feature manager

* add more tests

* partial fix for null values in table response format

* handle partial null fields

* comments for easy testing

* move frameName allocation in readSeries

* one less append operation

* performance improvement by making string conversion once

pkg: github.com/grafana/grafana/pkg/tsdb/influxdb/influxql
             │ stream2.txt │            stream3.txt             │
             │   sec/op    │   sec/op     vs base               │
ParseJson-10   314.4m ± 1%   303.9m ± 1%  -3.34% (p=0.000 n=10)

             │ stream2.txt  │             stream3.txt              │
             │     B/op     │     B/op      vs base                │
ParseJson-10   425.2Mi ± 0%   382.7Mi ± 0%  -10.00% (p=0.000 n=10)

             │ stream2.txt │            stream3.txt             │
             │  allocs/op  │  allocs/op   vs base               │
ParseJson-10   7.224M ± 0%   6.689M ± 0%  -7.41% (p=0.000 n=10)

* add comment lines

---------

Co-authored-by: İnanç Gümüş <m@inanc.io>

dmihai committed:
Auth: encrypt secrets for oauth providers in SSO settings API service (#79081)

encrypt secrets for oauth providers

yoziru committed:
Prometheus: Fix PromQail health check + add test (#77259)

promQail: fix health check + add test

eleijonmarck committed:
Team LBAC: Remove default rules of lbac rules (#77946)" (#78188)

This reverts commit 944593c4d542c5980e9c263ff46786b9ef1b7162.

Co-authored-by: Alexander Zobnin <alexanderzobnin@gmail.com>

dprokop committed:
Dashboard interactions reporting refactor, DashboardScene interactions handling (#79078)

* Refactor dashboard interactions tracking

* Local scenes tmp

* Use latest scenes

* Review refactor

* Fix enterprise tests

* Reporting parity for panel description

ashharrison90 committed:
Server Stats: Add skeleton loader (#79138)

* add server stat skeleton loader

* prevent flicker

ashharrison90 committed:
Teams: Add skeleton for teamlist (#79132)

add skeleton for teamlist

ashharrison90 committed:
Orgs: Add skeleton loader (#79141)

add skeleton for orgs list

kaydelaney committed:
Scenes: Add ability to change panel viz type (#78477)

* Scenes: Add ability to change panel viz type

---------

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

yuri-tceretian committed:
Alerting: Support for `condition` field in /api/v1/eval (#79032)


Co-authored-by: Sonia Aguilar <soniaaguilarpeiron@gmail.com>

gotjosh committed:
Alerting: Attempt to retry retryable errors (#79037)

* Alerting: Attempt to retry retryable errors

Currently in a draft state, but this was the minimal diff I could put together to exemplify how could achieve this.

Signed-off-by: gotjosh <josue.abreu@gmail.com>

---------

Signed-off-by: gotjosh <josue.abreu@gmail.com>

gotjosh committed:
Revert "Alerting: Attempt to retry retryable errors" (#79158)

Revert "Alerting: Attempt to retry retryable errors (#79037)"

This reverts commit 3e51cf09491e19442cdceb8e84c5fb3b9ef17e2c.

DanCech committed:
Storage: Unified Storage based on Entity API (#71977)

* first round of entityapi updates

- quote column names and clean up insert/update queries
- replace grn with guid
- streamline table structure

fixes

streamline entity history

move EntitySummary into proto

remove EntitySummary

add guid to json

fix tests

change DB_Uuid to DB_NVarchar

fix folder test

convert interface to any

more cleanup

start entity store under grafana-apiserver dskit target

CRUD working, kind of

rough cut of wiring entity api to kube-apiserver

fake grafana user in context

add key to entity

list working

revert unnecessary changes

move entity storage files to their own package, clean up

use accessor to read/write grafana annotations

implement separate Create and Update functions

* go mod tidy

* switch from Kind to resource

* basic grpc storage server

* basic support for grpc entity store

* don't connect to database unless it's needed, pass user identity over grpc

* support getting user from k8s context, fix some mysql issues

* assign owner to snowflake dependency

* switch from ulid to uuid for guids

* cleanup, rename Search to List

* remove entityListResult

* EntityAPI: remove extra user abstraction (#79033)

* remove extra user abstraction

* add test stub (but

* move grpc context setup into client wrapper, fix lint issue

* remove unused constants

* remove custom json stuff

* basic list filtering, add todo

* change target to storage-server, allow entityStore flag in prod mode

* fix issue with Update

* EntityAPI: make test work, need to resolve expected differences (#79123)

* make test work, need to resolve expected differences

* remove the fields not supported by legacy

* sanitize out the bits legacy does not support

* sanitize out the bits legacy does not support

---------

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

* update feature toggle generated files

* remove unused http headers

* update feature flag strategy

* devmode

* update readme

* spelling

* readme

---------

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

toddtreece committed:
Chore: Ensure BuildVersion is set when using CDN (#79169)

gotjosh committed:
Alerting: Attempt to retry retryable errors (#79161)

* Alerting: Attempt to retry retryable errors

Retrying has been broken for a good while now (at least since version 9.4) - this change attempts to re-introduce them in their simplest and safest form possible.

I first introduced #79095 to make sure we don't disrupt or put additional load on our customer's data sources with this change in a patch release. Paired with this change, retries can now work as expected.

There's two small differences between how retries work now and how they used to work in legacy alerting.

Retries only occur for valid alert definitions - if we suspect that that error comes from a malformed alert definition we skip retrying.
We have added a constant backoff of 1s in between retries.

---------

Signed-off-by: gotjosh <josue.abreu@gmail.com>

toddtreece committed:
K8s: Refactor metrics to share k8s registry (#79106)

ryantxu committed:
Chore: remove the entity kind registry (#79178)

ryantxu committed:
Storage: remove git backing for storage (#79181)

toddtreece committed:
K8s: Build openapi defs in production (#79182)

leeoniya committed:
Transformations: Add (base field name) to Join transformer (#79192)

dprokop committed:
DashboardScene: Add support for showing notices in panel header (#79146)

dprokop committed:
DashboardScene: Skeleton for Panel edit data tabs (#79130)

ashharrison90 committed:
Library panels: Add loading skeleton (#79087)

* add library panel card skeleton

* lineHeight: 1 instead of 0

joshhunt committed:
BrowseDashboards: Fix alerts folder view cards not stacking correctly (#79162)

leventebalogh committed:
Plugins: Suppress 403 preloading errors (#79154)

* fix: stop displaying error alerts when plugins cannot be loaded

* fix: only hide the error if it is a 403 error

* fix: revert unnecessary object spreading

ashharrison90 committed:
Snapshots: Add loading skeleton (#79134)

* add snapshot skeleton

* add some unit tests

joshhunt committed:
Chore: Deprecate FieldConfigEditorProps in favor of StandardEditorProps (#78319)

* Deprecate FieldConfigEditorProps in favor of StandardEditorProps

* update editors to no longer use FieldOverrideEditorProps

gabor committed:
mysql: refactor logs handling (#79148)

* mysql: refactor logs handling

* fixed unit test

idafurjes committed:
Fix capitalising Folder structs (#79207)

* Fix capitalising Folder structs

* Add swagger and fix api dashboard test

* Add swagger files with bingo installed

gabor committed:
postgres: refactor logs-handling (#79147)

* postgres: refactor logs-handling

* fixed unit test

marefr committed:
Upgrade sdk to v0.196.0 (#79021)

yangkb09 committed:
Chore: Remove FolderID from DTO Folder (#78420)

* Chore: Remove FolderID from DTO Folder

* chore: add OrgID field to an instance of SaveDashboardCommand

* chore: add another OrgID to pair with the FolderUID:

* chore: add OrgId to Folder struct and expectedParentOrgIDs to testCase struct, unsure if last part is necessary

* Fix folder test, add expected orgID

* chore: regen specs

---------

Co-authored-by: Ida Furjesova <ida.furjesova@grafana.com>

dprokop committed:
DashboardScene: Use query options for relative time range calculation of the panel embed url (#79220)

ashharrison90 committed:
Service accounts: Add skeleton loader (#79224)

* use emotion object syntax

* move create button to be a page action

* create and use skeleton

mgyongyosi committed:
Auth: Improve groups claim setup docs for AzureAD (#78791)

* Improve groups claim setup docs

* Improve the GroupMember.Read.All perm docs

* Apply suggestions from code review

* Update docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md

* Update docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md

* Update the groups and app roles documentation

* Update

* Split long list to separate sections

* fix anchor

* Apply suggestions from code review

* Address other comments, fix anchors

* Address other comments, lint

* Apply suggestions from code review

* Changes

* final adjustments

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: lwandz13 <126723338+lwandz13@users.noreply.github.com>

dprokop committed:
AnnotationsMarker: Fix non-null assertion operator usage (#79205)

yangkb09 committed:
Chore: Remove FolderId from DashboardMeta (#78501)

* Chore: Remove FolderId from DashboardMeta

* chore: replace FolderId with FolderUid

* chore: fix test by adding FolderUID

aleks-p committed:
Pyroscope: Improve label suggestions in query editor (#78861)

* Send sanitized selectors to the Pyroscope backend for LabelNames and LabelValues

* Clean LabelNames response to remove already used labels

* Improve performance after major changes

* Fix import order

* Further improve rendering performance

* Fix frontend tests

* Fix fake pyroscope client signature

* Bump pyroscope/api dependency to include start/end in LabelNames/LabelValues

* Fix issue with old queries running when using the run button

* Add generated file

* Make code more readable, add a few comments

* Format with prettier

* Fix error when assigning data

* Revert "Add generated file"

This reverts commit c4f33727b8db6b934d14a3a0dfb6f4ce4174a406.

* Remove leftover code

* Simplify query editor internal state objects

* Move label selector validation up, improve label filtering

* Simplify query editor state, switch to debounce to reduce rerenders

* Revert cosmetic change

Rperry2174 committed:
Flamegraph: Add table filtering for Flamegraph panel (#78962)

yuri-tceretian committed:
Alerting: Fix fine-grained rule access control to use 403 for authorization error (#79239)

* use 403 for authorization error
* update silences API
* add ForbiddenError to rule API responses

VikaCep committed:
Alerting: Fix the "Run queries" button to preview queries (#79252)

Fix the "Run queries" button to preview queries

iwysiu committed:
CloudWatch: remove usage of TimeSrv (#79096)

ashharrison90 committed:
Settings: Add skeleton loader (#79234)

* add skeleton for settings page

* go back to using style so it all overrides properly

joey-grafana committed:
Tempo: Update the embedded flame graph labelSelector (#79061)

* Update the embedded flame graph labelSelector

* Update labelSelector

mgyongyosi committed:
Auth: Use SSO settings service to load social connectors + refactor (#79005)

* Refactor to prevent cyclic dependencies

* Move list authorization to the API layer

* Init connectors using the SSO settings service in case the ssoSettingsApi feature toggle is enabled

* wip, need to handle the cyclic dep

* Remove cyclic dependency

* Align tests + refactor

* Move back OAuthInfo to social

* Delete pkg/login/social/constants

* Move reloadable registration to the social providers

* Rename connectors.Error to connectors.SocialError

yoziru committed:
Prometheus: improved PromQAIL v1 prompt (#79232)

* PromQail: update prompt and model

- Split up system and user prompt
- Use latest gpt-3.5-turbo-1106 model
- Add metric type to user prompt
- Simpler formatting for templates

* PromQail: fix fetching all available labels

Previously we were rendering the user selected templates. The prompt is expecting all available label keys for a given metric.

* use fetchSeriesLabels instead

* user prompt: fix trailing newline

academo committed:
Docs: Improve comments for panel props for plugins (#78596)

* Docs: Improve comments for panel props for plugins

* Update packages/grafana-data/src/types/panel.ts

Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

* Update packages/grafana-data/src/types/panel.ts

Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

---------

Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

vtorosyan committed:
Strip out auth token from shared URL when sharing a dashboard (#79262)

siddhikhapare committed:
Cloudwatch: Update error code metrics for ES/OpenSearch (#78990)

Update error code metrics

toddtreece committed:
K8s: Merge metric registries (#79256)

vtorosyan committed:
Service account: Update last used timestamp when token is used (#79254)

mgyongyosi committed:
Auth: Log a warning on unparsable team ids for GitHub OAuth  (#79261)

* Log warn if team ids are not integers

* Remove comment

dmihai committed:
Auth: check that oauthProvider is not nil in Logout() func (#79271)

check that oauthProvider is not nil

imatwawana committed:
Docs: add missing alt text (1/3) (#79274)

* Added missing alt text

* Reworded text

gabor committed:
mssql: prepare logs-handling for decouple-datasource changes (#79214)

mssql: refactor logs-handling

dprokop committed:
DashboardScene: Support custom actions in the nav toolbar (#79139)

* DashbaordScene: Support custom nav toolbar actions

* Move custom actions to scenes feature dir

* Review

* betterrerererererer

fabrizio-grafana committed:
Tempo: Fix read-only assignment (#79183)

samu6851 committed:
Update trace-integration.md (#77329)

dependabot[bot] committed:
Bump actions/stale from 8 to 9 (#79292)

Bumps [actions/stale](https://github.com/actions/stale) from 8 to 9.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/v8...v9)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ashharrison90 committed:
News: Add loading skeleton (#79009)

* convert to emotion object syntax

* add news skeleton

dependabot[bot] committed:
Bump google-github-actions/setup-gcloud from 1 to 2 (#79291)

Bumps [google-github-actions/setup-gcloud](https://github.com/google-github-actions/setup-gcloud) from 1 to 2.
- [Release notes](https://github.com/google-github-actions/setup-gcloud/releases)
- [Changelog](https://github.com/google-github-actions/setup-gcloud/blob/main/CHANGELOG.md)
- [Commits](https://github.com/google-github-actions/setup-gcloud/compare/v1...v2)

---
updated-dependencies:
- dependency-name: google-github-actions/setup-gcloud
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ashharrison90 committed:
Navigation: Independent docked state (#78954)

* initial start

* more progress

* behaviour working?

* only use media query when docked local storage state is true

* close menu when undocking

* remove unneeded animation code and fix focus when toggling between docked/undocked

* better feature toggle handling (can go back and forth)

* remove restoreFocus (for now)

svennergr committed:
Loki: Fix values variable with unexisting label and stream selector (#79310)

joey-grafana committed:
Pyroscope: Decouple from templateSrv (#79068)

* Decouple templateSrv

* Update test

academo committed:
Chore: Migrate from aria-label to data-testid in plugin-related tests (#79272)

* Update basicsettings for datasource

* Change aria-label for data-testid

* Update more tests

* Update betterer

* Remove unnecessary data-testid

* Put back aria label for some datasource selectors

* Try to use data-testid with alerts

* Update betterer

kaydelaney committed:
InspectDataTab: Use templateSrv.replace with getProcessedData (#79113)

eleijonmarck committed:
Auth: Fix anonymous user table useragent not parsable (#79307)

fix for useragent not parsable

Jguer committed:
Anon: Enable anon stats by default (#79312)

* enable anon stats by default

* update flag

soniaAguilarPeiron committed:
Alerting: Fix UI inheriting mute timings from parent when calculating the polic… (#79295)

* Fix UI inheriting mute timings from parent when calculating the policy tree properties

* Add test for this use case

svennergr committed:
Loki: Fix `timeRange` missing in variable requests (#79326)

itsmylife committed:
InfluxDB: Interpolate tag keys in influxql queries (#79288)

* interpolate tag keys

* fix unit tests

soniaAguilarPeiron committed:
Alerting: Fix for data source filter on cloud rules. (#79327)

* fix data source filter not worling for cloud rules

* Add test

imatwawana committed:
Docs: Add more detail about mapping olhc values (#78963)

toddtreece committed:
Chore: Bump k8s dependencies to v0.28.4 (#79331)

ashharrison90 committed:
Navigation: Fix showing command palette when menu is docked (#79335)

* fix showing command palette when menu is docked

* separate out command palette

DanCech committed:
Storage: Switch from tenant to namespace & remove GRN (#79250)

* remove GRN and switch tenant to namespace

* clean up remaining references

* simplify and remove inconsistency in With* parameters

* parse listing keys so we can use db index

* bump the schema version

---------

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

rwwiv committed:
Add Loki alert state history annotations feature toggle (#78970)

* Add no-op Loki alert state history annotations feature toggle

* Better feature toggle description

Co-authored-by: Alexander Weaver <weaver.alex.d@gmail.com>

alexweav committed:
alerting: Add tests for hooks (#79284)

Add tests for hooks

juanicabanas committed:
PublicDashboards: Revoke all public dashboards access to a specific user (#78709)

ryantxu committed:
K8s: Add resource type helper to avoid so many hardcoded names (#79344)

yuri-tceretian committed:
SSE: Add utility methods for HysteresisCommand  (#79157)

* add GetCommandsFromPipeline
* refactor method GetCommandType to func GetExpressionCommandType
* add function to create fingerprint frames
* add function to determine whether raw query represents a hysteresis command and a function to patch it with loaded metrics

yuri-tceretian committed:
Alerting: Add export of mute timings to file provisioning formats (#79225)

* add export of mute timings to file provisioning formats
* support export of mute timings to HCL

overvenus committed:
Heatmap: Fix null options migration (#79083)

* Fix heatmap null options migration

Some dashboard generation libraries, such as grafanalib, may produce
heatmaps with a null 'options' attribute. This condition triggers a
panic during migration, resulting in a blank heatmap panel.

This commit addresses the issue by returning default options if a panel has
a null 'options'.

Signed-off-by: Neil Shen <overvenus@gmail.com>

* Nullish coalescing panel.options

Signed-off-by: Neil Shen <overvenus@gmail.com>

---------

Signed-off-by: Neil Shen <overvenus@gmail.com>

IevaVasiljeva committed:
RBAC: change annotation scopes back (#79330)

Change the annotation scopes back to what they were

kalleep committed:
IDForwarding: Remove dev mode restriction for feature toggle (#79358)

Remove dev mode restriction for id forwarding feature toggle

sd2k committed:
Plugins: Add Command Palette extension point (#78098)

Co-authored-by: Marcus Andersson <marcus.andersson@grafana.com>

xnyo committed:
Plugins: Add hide_angular_deprecation setting (#79296)

* Add plugins.hide_angular_deprecation config ini

* Add more tests

* Remove the ability to use [plugins.*] hide_angular_deprecation

brendamuir committed:
Docs: Alerting descriptions for alerting-rules section (#79376)

Alerting descriptions for alerting-rules section

brendamuir committed:
Alerting docs: Adds Alerting descriptions for fundamentals section (#79383)

Adds Alerting descriptions for fundamentals section

ivanahuckova committed:
Loki: Decouple from Prometheus operationUtils (#78830)

* Loki: Decouple from Prometheus operationUtils

* Update comments

Jguer committed:
Anonymous: Add configurable device limit (#79265)

* Anonymous: Add device limiter

* break auth if limit reached

* fix typo

* refactored const to make it clearer with expiration

* anon device limit for config

---------

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

ashharrison90 committed:
Navigation: Update synthetics placement in IA (#79302)

* update synthetics placement in IA

* lowercase s in synthetics

brendamuir committed:
Alerting docs: revises descriptions for the manage notifications section (#79385)

ashharrison90 committed:
Skeleton: Abstract out attach/animation logic (#79309)

* apply styles globally to skeleton

* use abstraction everywhere

* just use withSkeleton

* add comment

* update docs

* use it in News as well

* rename withSkeleton to attachSkeleton

* move to @grafana/ui/src/unstable

* rename skeletonProps to rootProps

joey-grafana committed:
Pyroscope: Switch to sdk httpclient from infra httpclient (#79373)

Decouple httpclient

Commit:
I18n: Crowdin sync (#79386)

* New translations grafana.json (French)

* New translations grafana.json (Spanish)

* New translations grafana.json (German)

* New translations grafana.json (Chinese Simplified)

* Github Action: Auto-fix i18n files

---------

Co-authored-by: grafana-i18n-bot <grafana-i18n-bot@users.noreply.github.com>

joey-grafana committed:
Tempo: Update cheatsheet (#78797)

* Tempo: Update cheatsheet

* Update text

ivanahuckova committed:
Loki: Remove imported test utility function from Prometheus (#78901)

Loki: Remove import from prometheus test utility

fabrizio-grafana committed:
Tempo: Fix autocompletion with strings (#79370)

kaydelaney committed:
Dashboards: Run shared queries even when source panel is in collapsed row (#77792)

santihernandezc committed:
Chore: Simplify function signature for GetLatestAlertmanagerConfiguration (#79392)

julienduchesne committed:
`accesscontrol` swagger: Add `global` field to `RoleDTO` type (#79351)

* `accesscontrol` swagger: Add `global` field to `RoleDTO` type
The field is currently added in the MarshalJSON function so it isn't reflected in the spec
This PR sets the "static" version of the RoleDTO, that has the global field, as the swagger model

* Revert the marshalling logic

shabeebk committed:
Core: Rename index.ts to completionUtils.ts in Loki monaco completion provider (#79253)

Rename index.ts to completionUtils.ts in Loki monaco completion provider

This is a follow up to #76374. The name `index.ts` is not matching with
the pattern in the whole project as per the original discussion. It was
noted in https://github.com/grafana/grafana/pull/76080#discussion_r1352812657

brendamuir committed:
Alerting docs: remaining descriptions and deletes legacy topics (#79387)

Alerting docs: remaining descriptions and deletes legacy cloud alerting topics

ArturWierzbicki committed:
Storage: make it work for Postgres (#79378)

* comment out cockroachdb connection config option that does not work for postgres

* log the error instead

DanCech committed:
Storage: Add readme for storage server (#79366)

add readme for storage server

soniaAguilarPeiron committed:
Alerting: Fix max data points placeholder rendering decimals (#78946)

Fix max data points placeholder rendering decimals

doanbutar committed:
Special alerts for NoData and Error (#79193)

* Special alerts for NoData and Error

To avoid confusion why there is no additional label added when setting `Alerting` or `OK` in the No Data and Error handling

* ran prettier

---------

Co-authored-by: brendamuir <100768211+brendamuir@users.noreply.github.com>

academo committed:
Sandbox: Improve logging of sandbox lifecycle for monitoring (#79297)

* Sandbox: Log a plugin loading inside the sandbox

* Improve tracking of general sandbox errors

* Fix import

gotjosh committed:
Alerting: Refactor readiness check (#78799)

* Alerting: Refactor readiness check

Moves the readiness check to the mimir client and removes the need to assert that we have senders - it already has a queue and can hold notifications until we're ready to send them.
---------

Signed-off-by: gotjosh <josue.abreu@gmail.com>

santihernandezc committed:
Alerting: Fix auth for Nginx in Mimir Backend block (devenv) (#79404)

gelicia committed:
Explore: Create menu for short link button (#77336)

* WIP first pass

* Change to toolbarbutton/buttonselect pattern

* Move to two toolbarbuttons with dropdown

* Justify text to the right of icons

* Fix betterer

* Fix styling and tests

* Add to docs

* Perist last clicked, add translations

* move styling to core component

* use label for tooltip, let parser combine panes in multiple params

* Explore: Panes encoding suggestions (#78210)

panes encoding suggestions

* WIP-add menu groups

* Get group actions working

* add icons and non-local last selected state

* Fix translations after merge

* Add categories to translation, tweak the verbiage

* Fix translation extraction

* Fix tests

* fix translation conflict

* Log if time is absolute

---------

Co-authored-by: Giordano Ricci <me@giordanoricci.com>

zserge committed:
Chore: Move folder sync logic into one-shot migration (#78985)

Chore: move folder sync code into one-shot migration

knylander-grafana commented:
@fabrizio-grafana Do we need docs for this? it has the type/docs label on it, so I wanted to check in.

imatwawana committed:
Docs: add missing alt text - transformations 10.3 (2/3) (#79277)

* Added missing alt text

* Rebuilt markdown file

q
wq

* baldm0mma/081223-missing-alt-text-2/ add alt image content to images and build markdown

* baldmomma/081223-missing-alt-text-2/update alt text and build

* baldm0mma/081223-missing-alt-text-2/ make altText required

---------

Co-authored-by: jev forsberg <jev.forsberg@grafana.com>

santihernandezc committed:
Alerting: Check whether the internal Alertmanager is ready in remote secondary mode (#79406)

Alerting: Check whether the internal Alertmanager is ready in remote secondary

torkelo committed:
Stat: Fix inconsistent center padding (#79389)

sympatheticmoose committed:
update angular plugin list (#79380)

* rm scadavis

* add deprecation advice

* rm cloudflare

* rm opennms-helm

* rm other plugins

* update deprecation link

* make prettier happy

* Update docs/sources/developers/angular_deprecation/angular-plugins.md

* Update docs/sources/developers/angular_deprecation/angular-plugins.md

---------

Co-authored-by: Joseph Perez <45749060+josmperez@users.noreply.github.com>

julienduchesne committed:
ngalert swagger: Fix status code (#79415)

This endpoint returns a 202, not a 204
Let me know if we should instead change the response of the API

DanCech committed:
Storage: rename uid to name and name to title (#79418)

rename uid to name and name to title

imatwawana committed:
Docs: add/update frontmatter descriptions for visualizations pages (#79419)

Added or updated frontmatter descriptions

alexweav committed:
Alerting: Use mux router to match hooks, add support for path variables and methods (#79345)

* Use a router inside hooks rather than plain string matching

* Add test for mismatched method

DanCech committed:
Storage: raise errors when timestamps cannot be parsed (#79405)

* raise errors when timestamps cannot be parsed

* return partial origin info if timestamp parsing fails

leeoniya committed:
PanelQueryRunner: Shallow diff series DataFrames (#79257)

leeoniya committed:
OptionsPicker: Use fuzzy search and improve ranking of matches (#79286)

rwwiv committed:
Annotations/Alerting: Add Loki historian store stub (#78363)

* Add Loki historian store stub

* Add composite store

* Use composite store if Loki historian enabled

* Split store interface into read/write

* Make composite + historian stores read only

* Use variadic constructor for composite

* Modify Loki store enable logic

* Use dskit.concurrency.ForEachJob for parallelism

kevinwcyu committed:
Cloudwatch: Fix errors while loading queries/datasource on Safari (#79417)

* pass super.query to query runners

* fix types in tests

* clearer name for query function

* fix test

soniaAguilarPeiron committed:
Alerting: Add export mute timings feature to the UI. (#79395)

* Add export for all muteTimings and a single muteTiming

* Add test

* Fix snapshot

* Fix test

* Add mute timing name in file name when exporting single mute timing

xnyo committed:
Plugins: Add forward_host_env_vars setting (#79333)

* Plugins: Add forward_host_env_vars_plugins setting

* Renamed forward_host_env_vars_plugins to forward_host_env_vars

* Add readPluginIDsList

* refactoring

* lint

* Use util.SplitString

sd2k committed:
Plugins: Add Grafana config to app plugin context (#79439)

Jguer committed:
Auth: Update okta docs (#79381)

* update okta docs

* fix missing corrections

* fix missing corrections

mdvictor committed:
Table: Highlight row on shared crosshair (#78392)

* bidirectional shared crosshair table WIP

* add shared crosshair to table panel

* lower around point threshold

* add feature toggle

* add index based verification

* add adaptive threshold

* switch to debounceTime

* lower debounce to 100

* raise debounce back to 200

* revert azure dashboard

* re-render only rows list on data hover event

* further break down table component

* refactor

* raise debounce time

* fix build

ashharrison90 committed:
Data sources: Add skeleton loader (#79016)

* refactor out DataSourcesListCard

* add skeleton

* increase gap between buttons on skeleton to match real component

* lineHeight: 1 instead of 0

* refactor out ternary

yangkb09 committed:
Chore: Remove folderID from dashboardIdentity tests (#79195)

tskarhed committed:
Betterer: Add rule for gf-form (#79303)

Add betterer rule for gf-form

yangkb09 committed:
Chore: Remove FolderID from Dashboard Import (#78832)

* Chore: Remove FolderID from Dashboard Import

* chore: regen specs

* Remove OrgID from DashboardImportRequest and DashboardImportResponse

* Remove OrdIDs from swagger and tests

---------

Co-authored-by: Ida Furjesova <ida.furjesova@grafana.com>

ashharrison90 committed:
Revert "Betterer: Add rule for gf-form (#79303)" (#79445)

This reverts commit 7a38a5ab1d9469fb03a3ef2f2c594f8461cc4bd7.

juanicabanas committed:
PublicDashboards: Remove unsupported datasource warning for cloudwatch (#79403)

santihernandezc committed:
Alerting: Add time-based convergence in remote secondary mode (#78809)

* Alerting: Add a sync interval for ApplyConfig in remote secondary mode

* add routine to sync states and configs

* pass a cancellable context to syncRoutine(), remove tests for ApplyConfig, cache last config in memory

* extract logic to update config and state in the remote Alertmanager

* get latest config from the database

* avoid using separate goroutine for updating state and config

* clean up PR

* refactor, comments, tests

* update tests

* add config struct for remote secondary forked Alertmanager

* use errgroups for sync operations

* use waitgroup instead of errgroup

* remove helper method to sync AMs

* check for errors instead of bool syncErr

svennergr committed:
Elasticsearch: Fix modify query with backslashes (#79430)

* Elasticsearch: Fix modify query with backslashes

* add test devenv

gtk-grafana committed:
Loki: strip out invalid options for logs/metrics queries  (#79409)

* fix bug in split query where metric queries with maxLines of 0 would not execute

matyax committed:
Loki/Elasticsearch: prevent errors in onDashboardLoadedHandler (#79451)

* Loki: prevent errors in onDashboardLoadedHandler

* Elasticsearch: prevent errors in onDashboardLoadedHandler

eledobleefe committed:
TimeRangePicker: do not swap time ranges when `to` is before `from` (#78915)

JoaoSilvaGrafana committed:
Card: Revert grid template row size change (#79457)

kevinwcyu committed:
CloudWatch: Remove core imports from app/features/variables/types (#79361)

* CloudWatch: Remove core imports from app/features/variables/types

* fix import

* lint

GrafanaWriter committed:
Docs: Update list of open source projects (#79432)

* Docs: Update list of open source projects

Update to change Phlare to Pyroscope, add Faro, Beyla, Agent, k6, OnCall

* Ran prettier

---------

Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>

tolzhabayev committed:
Chore: Proposing a change of go-jose ownership (#79449)

changing go-jose ownership

itsmylife committed:
InfluxDB: Update SQL language url specification (#79408)

* accept urls with or without port

* remove redundant comment

* remove redundant function

fabrizio-grafana committed:
Merge branch 'main' into otp/decouple-tempo-frontend

julienduchesne committed:
ngalert openapi: Add `X-Disable-Provenance` to missing operations (#79278)

Swagger(ngalert): Add `X-Disable-Provenance` to missing operations
I added all functions that call the `determineProvenance` function

Schema changes are from:
`make` in `pkg/services/ngalert/api/tooling`
`make swagger-clean && make openapi3-gen` in root

adrapereira committed:
Tempo: Fix cache in TraceQL editor (#79468)

Fix 77380

ashharrison90 committed:
Navigation: Update frontend and app o11y icons  (#79454)

* use frontend o11y icon when docked

* update app o11y icon

imatwawana committed:
Docs: add/update frontmatter descriptions for panel config pages  (#79412)

* Added and updated frontmatter descriptions

* baldm0mma/121223-update-frontmatter-descriptions-2/ add description to template and build

* Updated description

Co-authored-by: Nathan Marrs  <nathanielmarrs@gmail.com>

* Updated description

---------

Co-authored-by: jev forsberg <jev.forsberg@grafana.com>
Co-authored-by: Nathan Marrs <nathanielmarrs@gmail.com>

santihernandezc committed:
Alerting: Send configuration and state to the remote Alertmanager on shutdown (#78682)

* Alerting: Send configuration and state to the remote Alertmanager on shutdown

* Alerting: Add a sync interval for ApplyConfig in remote secondary mode

* add routine to sync states and configs

* pass a cancellable context to syncRoutine(), remove tests for ApplyConfig, cache last config in memory

* extract logic to update config and state in the remote Alertmanager

* get latest config from the database

* avoid using separate goroutine for updating state and config

* clean up PR

* refactor, comments, tests

* update tests

* remove canceled context from calls to StopAndWait()

* create context with timeout and send config and state to remote Alertmanager

* update tests

* address code review comments

adela-almasan committed:
Tooltip: Improved Timeseries and Candlestick tooltips (#75841)

nmarrs committed:
gdev: Add gdev panels for enum feature (#78350)

Co-authored-by: drew08t <drew08@gmail.com>

nmarrs committed:
BarChart: Fix incorrect legend field labels (#78511)

Co-authored-by: Leon Sorokin <leeoniya@gmail.com>

khushijain21 committed:
Chore: Remove gf-form in UserPermission & Annotation Query (#78785)

* Chore: Remove gf-form in UserPermission

* more changes

* Update public/app/features/admin/UserPermissions.tsx

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

* more change

* Added annotationQueryEditor

* empty div

* Update public/app/plugins/datasource/elasticsearch/components/QueryEditor/AnnotationQueryEditor.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* final

---------

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>
Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

gabor committed:
sql: use logging functionality from plugin-sdk (#79294)

svennergr committed:
Loki: Add timeRange to labels requests in LogContext to reduce loading times (#79478)

Loki: Add timeRange to labels requests

IevaVasiljeva committed:
RBAC: Extend the docs with an example of mapping to a fixed role (#79444)

extend the docs with an exmaple of mapping to a fixed role

kaydelaney committed:
Templating: Json interpolation of single-value default selection does not create valid json (#79137)

aangelisc committed:
Plugins: Include Azure settings as a part of Grafana config sent in plugin requests (#79342)

* Add Azure settings and update tests

* Filter by plugin ID

* Add forward settings config variable

* Update line

* Add tests

* Update so that data sources are fully defined in config

* Update SDK and test

* Fix lint

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Andrew Hackmann <5140848+bossinc@users.noreply.github.com>

* Remove unnecessary if

---------

Co-authored-by: Andrew Hackmann <5140848+bossinc@users.noreply.github.com>

axelavargas committed:
Dashboard: Migration - Dashboard Settings Variables (List, Duplicate, Delete) (#78917)

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

soniaAguilarPeiron committed:
Alerting: Fix queries and expressions in rule view details. (#79497)

* fix queries and expressions in rule-view details

* Add test

sarahzinger committed:
Cloudwatch: Update cloudwatchNewRegionsHandler to General Availability. (#79348)

* Cloudwatch: move cloudwatchNewRegionsHandler to General Availability.

toddtreece committed:
Chore: Bump k8s dependencies to v0.29.0 (#79492)

* Chore: Bump k8s dependencies to v0.29.0

* update the openapi fork

* use post process spec

---------

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

PoorlyDefinedBehaviour committed:
Plugins: Add option to disable TLS in the socks proxy (#79246)

* Plugins: add option to disable TLS in the socks proxy

* fix allow_insecure docs

* upgrade github.com/grafana/grafana-plugin-sdk-go from v0.196.0 to v0.197.0

* fix conflicts

kalleep committed:
Analytics: Return empty analytic settings for users that are not signed in (#79515)

Return empty analytic settings for users that are not signed in

wbrowne committed:
Plugins: Make renderer service load renderer plugin (#77854)

* rendering service loads renderer plugin

* update naming

* tidy

* apply PR feedback

* fix missing feature manager

* fix step

* set plugin

rwwiv committed:
Alerting: Add rule ID and title to alert state history Loki entry (#79481)

* Add rule ID and title to Loki entry

* Combine related tests

toddtreece committed:
Chore: Remove replace for kube-openapi (#79533)

owensmallwood committed:
Config: Can add static headers to email messages (#79365)

* Can add allowed custom headers to an email Message. WIP.

* adds slug as a custom email header to all outgoing emails

* Headers are static - declared as key/value pairs in config. All static headers get added to emails.

* updates comment

* adds tests for parsing smtp static headers

* updates test to assert static headers are included when building email

* updates test to use multiple static headers

* updates test names

* fixes linting issue with error

* ignore gocyclo for loading config

* updates email headers in tests to be formatted properly

* add static headers first

* updates tests to assert that regular headers like From cant be overwritten

* ensures only the header is in a valid format for smtp and not the value

* updates comment and error message wording

* adds to docs and ini sample files

* updates smtp.static_headers docs examples formatting

* removes lines commented with semi colons

* prettier:write

* renames var

imatwawana committed:
Docs: add/update frontmatter descriptions for What's new, upgrade guide, breaking changes pages (#79425)

Added and updated frontmatter descriptions

imatwawana committed:
Docs: add/update frontmatter descriptions for Dashboards pages (#79357)

* Added and updated frontmatter descriptions for dashboards pages

* fixed typo

* Updated public dashboards page description

lucychen-grafana committed:
Task: Improve public dashboard config url formatting (#79424)

* Update publicdashboard config url

* add slug to sessiondashboard

* swagger doc update

imatwawana committed:
Docs: update description and add alt text (#79538)

* Edited page description and added missing alt text

* Rebuilt markdown file

imatwawana committed:
Docs: add missing alt text (#79537)

Added missing alt text

khushijain21 committed:
Chore: Removes gf-form from admin/user_ (#79502)

* userProfile

* userpermis

* usersession

* useradmin

joey-grafana committed:
Tempo: Allow usage of template vars with service map query (#79382)

* Apply template vars to serviceMapQuery

* Add test

tskarhed committed:
Chore: Remove gf-form and change to CSS object in Graphite function editor (#79446)

Chore: remove gf-form and add CSS fumction object

tolzhabayev committed:
Chore: Upgrade Go to 1.21.5 (#79329)

* bumping grafanas go version to 1.21.5

* adding how to upgrade go doc

* removing the whole ci-build and corresponding pipelines

* clarifying that we have to run make drone

* fixing starlark linter and removing unused code

* Adding note about enterprise

* trying to change the underyling image to avoid musl issues

* only need to golang one

Jguer committed:
Access: GA split scope (#79555)

* GA split scope

* fix tests

mgyongyosi committed:
Chore: Configure SkipOrgRoleSync from OAuthInfo for OAuth connectors (#79443)

* Configure SkipOrgRoleSync from OAuthInfo

* Remove skipOrgRoleSync from socialbase and connectors

* Add test to socialimpl.ProvideService

* Deprecate AuthSettings' fields

* clean up misleading init of frontendsettings.Auth

tskarhed committed:
Betterer: Add our own regexp matcher (#79448)

* Add betterer rule for gf-form

* Betterer: Write generic regexp matcher

* Add space

* Update lockfile

* Update comments

* Add comment with notice

oscarkilhed committed:
Transformations: Move regression analysis transformation to private preview (#79500)

* move regression analysis transformation to private preview

* fix case

oscarkilhed committed:
Transformations: Move formatString transformation to private preview (#79498)

* Move moving average and cumulative sum to private preview

* update docs

* move formatString to private preview

* rebuild docs

* undo changes that don't belong to this commit

* undo cumulative/window featureflag

* fix case

joey-grafana committed:
Tempo: Fix graph dragging (#79508)

* Fix graph dragging

* Add comment

ivanortegaalba committed:
GeneralSettings: hide time picker from settings using scenes (#79526)

fabrizio-grafana committed:
Tempo: Support special characters in identifiers (#79565)

mgyongyosi committed:
Chore: Cleanup SocialBase + connectors and use the OAuthInfo (#79453)

* Clean up SocialBase

* Cleanup other connectors

eledobleefe committed:
Explore: Avoid swapping time range when `from` value is after `to` value (#79520)

mgyongyosi committed:
OAuth: Remove accessTokenExpirationCheck feature toggle (#79455)

* Remove accessTokenExpirationCheck from code and align docs

* Apply suggestions from code review

* lint

---------

Co-authored-by: lwandz13 <126723338+lwandz13@users.noreply.github.com>

joey-grafana committed:
Tempo: Carry over query from Search tab to TraceQL tab (#79442)

* Carry over query from Search tab to TraceQL tab

* Update tracking

* Update flow

* Updates to logic

fabrizio-grafana committed:
Merge branch 'main' into otp/decouple-tempo-frontend
